public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -286,7 +299,10 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
         text: `You have unsaved changes to this dashboard. Are you sure you want to discard them?`,
         icon: 'trash-alt',
         yesText: 'Discard',
-        onConfirm: this.exitEditModeConfirmed.bind(this),
+        onConfirm: () => {
+          this.exitEditModeConfirmed();
+          this._scopesFacade.exitViewMode();
```

Comment:
this is leaking edit mode specific behavior. Don't think this should be facade's public interface. I would expect this to become:
```ts
this.getScopesFacade().disable()
```

Reply:
Maybe even `getScopesFacade` is just a util that accepts current scene object and returns the closest behavior:

```ts
function getScopesFacade(obj: SceneObject): ScopesFacade | null;
```

Reply:
Done!